php/libraries/Project.class.inc

```diff
@@ -227,13 +227,16 @@ class Project
     /**
      * Set this project's name
      *
-     * @param string $projectName This project's name
+     * @param ?string $projectName The new name. Uses class property when null.
      *
      * @return void
      */
-    public function updateName(string $projectName): void
+    public function updateName(?string $projectName = null): void
```

Comment:
Why did you add nullable here? Do we want the project name to be null?

Reply:
it defaults to the objects _projectName which is required by the singleton. the reason it is nullable here is if you do something like
```
setName("BLA");
updateName();
```
that would persist the name to BLA

Reply:
Why not use an empty string instead of null?

Reply:
not sure i understand, whats the advantage of an empty string ?

Reply:
In terms of static analysis it's a bit better to have functions that deal strictly with strings if they're meant to be strings instead of `string|null`. If something can be achieved with a stricter type it's usually helpful.

I would flip it around and say what is the advantage of `null` when we can just use an empty string?

Reply:
i see you point for the static analyses.

there are 2 things here I wanna make sure not to confuse

 - the nullable `?` is because given the current logic of the class, that would allow people the set the value and then choose the persist it by calling the object function with no arguments. i would hate seeing anywhere is the code `updateName(null)` or `updateName('')`
- the `= null` part, is just because i prefer doing null checks then empty string checks when possible. in this specific case it doesnt make a difference since the empty function returns true to both. So i can change it to `''`

Reply:
Yep that sounds good to me. Like I said I thing `string $projectName = ''` is a bit better to `?string $projectName = null` but in this case specifically they're logically equivalent and either works.